src/core/lib/event_engine/posix_engine/posix_engine.cc

```diff
@@ -316,6 +317,8 @@ void PosixEnginePollerManager::Run(absl::AnyInvocable<void()> cb) {
 }
 
 void PosixEnginePollerManager::TriggerShutdown() {
+  GPR_DEBUG_ASSERT(trigger_shutdown_called_ == false);
```

Comment:
Can be `GPR_DEBUG_ASSERT(!trigger_shutdown_called_);`

Or you can merge these two lines into `GPR_ASSERT(!std::exchange(trigger_shutdown_called_, true));`, but it's debatably less readable.

Reply:
I opted for the readability since this is a string we could be reading in a test failure at some point.